### PR TITLE
fix[StatusChatMenu]: updated menu options as per designs

### LINKED ISSUE #13393: Close Chat and Clear History flows updated to align with mobile app in terms of functionality and terminology
The `Close Chat` and `Clear History` context menu items for 1:1 chat have been updated to align with mobile.

Screenshot 2024-02-01 at 9 12 25 PM

In addition the confirmation dialogs for each action have also been updated to align with mobile:

**Close chat:**

Screenshot 2024-02-01 at 9 14 08 PM

Once the chat is closed, the chat disappears from the user's Chat list and the next chat down in the list is selected. If the user is closing their only chat, the empty state for the Messaging app is shown.

--

**Clear chat history:**
Note that there are 2 checkboxes:
`Delete all messages on my side`- mandatory (on and disabled)
`Delete my messages for [counterparty name]` - optional - if checked, this will delete all of the actioning user's messages from the chat on the counterparty side.

Screenshot 2024-02-01 at 9 14 33 PM

Figma:
https://www.figma.com/file/Mr3rqxxgKJ2zMQ06UAKiWL/%F0%9F%92%AC-Chat%E2%8E%9CDesktop?type=design&node-id=24942-215153&mode=design&t=GWTxxV5I2KBpua8E-4

### PR DESCRIPTION
**Pull request by alexandraB99**
The "Delete my messages for viola.stateofus.eth" option is not yet supported, TBD [here](https://github.com/status-im/status-desktop/issues/14169).
Fetching messages for specific timeframes TBD [here](https://github.com/status-im/status-desktop/issues/14312)
Leaving and deleting group when being the owner TBD [here](https://github.com/status-im/status-desktop/issues/14311)

Closes #13393 
Depends on https://github.com/status-im/status-go/pull/5813

### What does the PR do
StatusChatMenu: updated menu options as per designs

### Affected areas
StatusChatMenu

### Screenshot of functionality (including design for comparison)

1-1 chat:
image

Group chat:
image

Community chat, admin:
image

Community chat, non-admin:
image

### CODE REVIEW

File: ui/imports/shared/views/chat/ChatContextMenuView.qml
Comment (caybro) [inline ui/imports/shared/views/chat/ChatContextMenuView.qml:129]: I'm sure this has to stay, @igor-sirotin @jrainville

File: ui/imports/shared/controls/chat/menuItems/MuteChatMenuItem.qml
Comment (caybro) [inline ui/imports/shared/controls/chat/menuItems/MuteChatMenuItem.qml:35]: Did we really lose the "7 days" period from the popup menu?

File: ui/imports/shared/views/chat/ChatContextMenuView.qml
Comment (igor-sirotin) [reply ui/imports/shared/views/chat/ChatContextMenuView.qml:129]: Indeed, why did this go away 😢 
It's under `showDebugOptions`, so won't bother most users.

**Review by igor-sirotin — CHANGES_REQUESTED** [review]
Please bring back the debug actions 🙂

**Comment by jrainville** [comment]
@alexjba do you want someone from the core team to take over this PR?

**Comment by alexjba** [comment]
> @alexjba do you want someone from the core team to take over this PR?

Yes, thanks! Not sure we need it anymore

**Comment by caybro** [comment]
@jrainville @igor-sirotin pls review :)

File: ui/imports/shared/views/chat/ChatContextMenuView.qml
Comment (jrainville) [inline ui/imports/shared/views/chat/ChatContextMenuView.qml:231]: This one seems duplicated from the one above. One is for OneToOne and the other for other types of chat, but I don't see any other differences. Is it just about the positioning?

**Review by jrainville — COMMENTED** [review]
Looks good. There is just a small duplication maybe?

File: ui/imports/shared/views/chat/ChatContextMenuView.qml
Comment (caybro) [reply ui/imports/shared/views/chat/ChatContextMenuView.qml:231]: Exactly, and I see no other way around it

**Comment by caybro** [comment]
> Please bring back the debug actions 🙂 

Done 🙂

File: ui/imports/shared/views/chat/ChatContextMenuView.qml
Comment (jrainville) [reply ui/imports/shared/views/chat/ChatContextMenuView.qml:231]: I looked at the designs and it is my humble opinion that the "close chat" and "leave group" should both be at the bottom.
It makes it more standardized and also makes it so we could avoid the duplication :smile: 
I'll let @benjthayer decide

Comment (caybro) [reply ui/imports/shared/views/chat/ChatContextMenuView.qml:231]: Yeah, so I didn't change that one from the original PR... at the same time, I wasn't able to find out the difference between a one-to-one chat and otehr chat types

**Comment by caybro** [comment]
@jrainville @benjthayer updated the screenshots in this PR to show the various menu options; pls advise if there's anything to change

**Review by igor-sirotin — APPROVED** [review]

**Comment by benjthayer** [comment]
> @jrainville @benjthayer updated the screenshots in this PR to show the various menu options; pls advise if there's anything to change (you can ignore the "Debug options", these are not shown to non-dev users)

LGTM @caybro thanks!

Only one I see missing tho is Owner of a Group Chat (additional option to `Leave and Delete Group`):

Screenshot 2024-09-09 at 11 06 25

**Comment by jrainville** [comment]
> > > @jrainville @benjthayer updated the screenshots in this PR to show the various menu options; pls advise if there's anything to change (you can ignore the "Debug options", these are not shown to non-dev users)
> > 
> > 
> > LGTM @caybro thanks!
> > Only one I see missing tho is Owner of a Group Chat (additional option to `Leave and Delete Group`):
> 
> Ah right, that one is indeed missing but I don't think we've implemented that option yet on the backend. @jrainville: any idea? Otherwise I think it's good to go

There is another issue for that, but there is no consensus for that and I don't think it's worth working on that now.

IMO, we can go with just those changes for now.

**Comment by igor-sirotin** [comment]
Waiting for this to be merged: https://github.com/status-im/status-go/pull/5813

**Comment by benjthayer** [comment]
> > > > @jrainville @benjthayer updated the screenshots in this PR to show the various menu options; pls advise if there's anything to change (you can ignore the "Debug options", these are not shown to non-dev users)
> > > 
> > > 
> > > LGTM @caybro thanks!
> > > Only one I see missing tho is Owner of a Group Chat (additional option to `Leave and Delete Group`):
> > 
> > 
> > Ah right, that one is indeed missing but I don't think we've implemented that option yet on the backend. @jrainville: any idea? Otherwise I think it's good to go
> 
> There is another issue for that, but there is no consensus for that and I don't think it's worth working on that now.
> 
> IMO, we can go with just those changes for now.

Cool! No worries. The `Leave and Delete Group` just ensures a group chat is not orphaned without an owner. Not a massive issue for now - but perhaps a security consideration for later.

**Comment by caybro** [comment]
@jrainville rebased and updated to point to the correct status-go branch

**Review by jrainville — APPROVED** [review]

**Pull request merged**
